stix_shifter_modules/cbcloud/stix_transmission/status_connector.py

```diff
@@ -0,0 +1,74 @@
+import json
+
+from enum import Enum
+from stix_shifter_utils.modules.base.stix_transmission.base_status_connector import BaseStatusConnector
+from stix_shifter_utils.modules.base.stix_transmission.base_status_connector import Status
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+
+
+class CbCloudStatus(Enum):
+    # WAIT, EXECUTE, SORTING, COMPLETED, CANCELED, ERROR
+    RUNNING = 'RUNNING'
+    COMPLETED = 'COMPLETED'
+    CANCELED = 'CANCELED'
+    ERROR = 'ERROR'
+
+
+class StatusConnector(BaseStatusConnector):
+    def __init__(self, api_client):
+        self.api_client = api_client
+        self.logger = logger.set_logger(__name__)
+
+    # Map data source status to connector status
+    def __getStatus(self, status):
+        switcher = {
+            CbCloudStatus.RUNNING.value: Status.RUNNING,
+            CbCloudStatus.COMPLETED.value: Status.COMPLETED,
+            CbCloudStatus.CANCELED.value: Status.CANCELED,
+            CbCloudStatus.ERROR.value: Status.ERROR
+        }
+        return switcher.get(status).value
+
+    def create_status_connection(self, search_id):
+        response = self.api_client.get_search_status(search_id)
+        response_code = response.code
+        response_text = response.read()
+        error = None
+        response_dict = dict()
+
+        try:
+            response_dict = json.loads(response_text)
+        except ValueError as ex:
+            self.logger.debug(response_text)
+            error = Exception(f'Can not parse response: {ex}')
+
+        # Based on the response
+        # return_obj['success'] = True or False
+        # return_obj['status'] = One of the statuses as defined in the Status class:
+        # Status.RUNNING, Status.COMPLETED, Status.CANCELED, Status.ERROR
+        # return_obj['progress'] = Some progress code if returned from the API
+
+        # Construct a response object
+        return_obj = dict()
+        return_obj['success'] = False
+
+        if response_dict and response_code == 200:
+            completed = response_dict['completed']
+            contacted = response_dict['contacted']
+            return_obj['success'] = True
+            return_obj['progress'] = {
```

Comment:
nested json for `progress` property will throw error to the caller as the caller of the status endpoint is expecting just the progress number. you can do the following instead-

```
return_obj['progress'] = completed
return_obj['contacted'] = contacted
```

Reply:
Perfect, that's an easy fix.

Reply:
This has been updated per the suggestion.